.github/workflows/django-overhead-profile.yml

```diff
@@ -0,0 +1,96 @@
+name: django-overhead-profile
+on:
+  push:
+    branches: [master]
+  pull_request:
+jobs:
+  django-overhead-profile:
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: trace-examples/python/django/django-simple
+    env:
+      DJANGO_ALLOWED_HOSTS: 127.0.0.1
+      DJANGO_SECRET_KEY: vNIF0IeM1IN9YAOOB8yEmNddSvnnXxx7E7D3TwWsdno
+      DATABASE_URL: sqlite:///django.db
+      DJANGO_SETTINGS_MODULE: config.settings.production
+      USE_DOCKER: false
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          path: ddtrace
+
+      - uses: actions/checkout@v2
+        with:
+          repository: DataDog/trace-examples
+          ref: 760deae1fd2f2371cf813d3ff3ca9f0e040e8c60
+          path: trace-examples
+
+      - uses: actions/setup-python@v2
+        with:
+          python-version: "3.8"
+
+      - name: Prepare Django app
+        run: |
+          pip install -r requirements/production.txt
+          python manage.py migrate
+
+      - name: Install ddtrace
+        run: pip install -e $GITHUB_WORKSPACE/ddtrace
+
+      - name: Checkout Austin development branch
+        uses: actions/checkout@v2
+        with:
+          repository: P403n1x87/austin
+          ref: b029cb7728783cc1035a032c9689f272e6ca2de6
+          path: austin
+
+      - name: Compile Austin on Linux
+        run: |
+          cd $GITHUB_WORKSPACE/austin
+          gcc -Wall -O3 -Os -s -pthread src/*.c -o src/austin
+          cd -
+
+      - name: Install diff tool dependencies
+        run: |
+          pip install rich
+          pip install git+git://github.com/p403n1x87/austin-python.git@acf3f3d3263a583c30f5c95e4b3ad337a6793aa3
+
+      - name: Install k6
+        run: |
+          mkdir $GITHUB_WORKSPACE/k6
+          cd $GITHUB_WORKSPACE/k6
+          curl https://github.com/loadimpact/k6/releases/download/v0.26.2/k6-v0.26.2-linux64.tar.gz -L | tar xvz --strip-components 1
+
+      - name: Collect profiles
+        run: |
+          mkdir ${{ github.workspace }}/artifacts
+
+          gunicorn config.wsgi --pid gunicorn.pid &
+          sleep 1
+          $GITHUB_WORKSPACE/k6/k6 run --out json=artifacts/baseline-load.json $GITHUB_WORKSPACE/ddtrace/scripts/django-overhead-profile-load.js &
+          sleep 1
+          sudo $GITHUB_WORKSPACE/austin/src/austin -i 200 -o ${{ github.workspace }}/artifacts/baseline.austin -sp $(cat gunicorn.pid) -x 60
```

Comment:
Why aren't using our own profiler to gather profiles?

Reply:
The idea here is to sample at high frequency for a set number of seconds to get as a high res picture as possible. If we can do that with our own profiler then we can certainly use that instead and simplify this script. One other thing we do is build diff, so perhaps we should also consider how easily we can do that with the profiler.

Reply:
if we want really high resolution, should we just use cProfile?

Reply:
Another motivation for this choice is so we can collect profiles for the baseline without using ddtrace.

Reply:
> if we want really high resolution, should we just use cProfile?

I think cProfile could complement the picture by providing a deterministic snapshot of what's happening

Reply:
> The idea here is to sample at high frequency for a set number of seconds to get as a high res picture as possible. If we can do that with our own profiler then we can certainly use that instead and simplify this script. One other thing we do is build diff, so perhaps we should also consider how easily we can do that with the profiler.

I think we can do that with ddtrace's `Profiler`. Happy to provide guidance. The only requirement would be to have this test run long enough to have statically meaningful data.

You can use env var `DD_PROFILING_OUTPUT_PPROF` to dump the pprof files locally rather upload them.
You can set `DD_PROFILING_MAX_TIME_USAGE_PCT=100` to make sure the 100 Hz freq is respected.
You could go faster than that; in theory up to `1/sys.getswitchinterval()` which would be 200 Hz by default. That's not supported by the code but it'd be easy to add.

> Another motivation for this choice is so we can collect profiles for the baseline without using ddtrace.

Well, you can use the profiler without the tracer. The biggest problem is that indeed you'd load the `Tracer` and all its code, but that's an old design issue of the lib unfortunately 😢